arthur_bench/scoring/summary_quality.py

```diff
@@ -152,21 +136,83 @@ def run(
                 f"Truncated {num_truncated} out of {len(inputs)} total summary inputs "
                 f"to {self.context_window} characters"
             )
+        return truncated_inputs
 
-        return super().run(
+    async def arun(
+        self,
+        candidate_outputs: List[str],
+        reference_outputs: Optional[List[str]] = None,
+        inputs: Optional[List[str]] = None,
+        contexts: Optional[List[str]] = None,
+        batch_size: int = ASYNC_BATCH_DEFAULT,
+    ) -> Union[List[float], List[ScoreResult]]:
+        if inputs is None:
+            raise UserValueError(
+                "input text is required for this scorer. Please provide a dataframe "
+                "column or a list of your "
+                "input text strings in the Test Suite."
+            )
+        if reference_outputs is None:
+            raise UserTypeError(
+                "Reference Outputs must be provided for Summary Quality scorer. Please "
+                "provide reference outputs to the test suite"
+            )
+
+        if contexts is not None:
+            raise UserValueError(
+                "using context is not currently supported for summary quality"
+            )
+        truncated_inputs = self._truncate_inputs(
+            inputs, reference_outputs, candidate_outputs
+        )
+
+        return await super().arun(
             candidate_outputs, reference_outputs, truncated_inputs, contexts, batch_size
         )
 
-    def run_batch(
+    def run(
         self,
+        candidate_outputs: List[str],
+        reference_outputs: Optional[List[str]] = None,
+        inputs: Optional[List[str]] = None,
+        contexts: Optional[List[str]] = None,
+        batch_size: int = SINGLE_ITEM_BATCH_DEFAULT,
+    ) -> Union[List[ScoreResult], List[float]]:
+        truncated_inputs = self._truncate_inputs(
+            inputs, reference_outputs, candidate_outputs
+        )
+
+        return super().run(
+            candidate_outputs, reference_outputs, truncated_inputs, contexts, batch_size
+        )
+
+    def _parse_response(self, response: Dict[str, Any]) -> ScoreResult:
+        score = None
+        if "text" in response:
+            llmchoice = response["text"][:3]
+            score = LLM_CHOICE_TO_FLOAT.get(llmchoice)
+            if score is not None:
+                result = ScoreResult(
+                    score=score,
+                    category=LLM_CHOICE_TO_CATEGORIES.get(
+                        llmchoice, LLM_CHOICE_TO_CATEGORIES["default"]
+                    ),
+                )
+
+        # return -1.0 if the LLMChain returns an invalid result
+        if score is None:
+            result = ScoreResult(
+                score=-1.0, category=LLM_CHOICE_TO_CATEGORIES["default"]
+            )
```

Comment:
For readability, I think this is equivalent and a bit easier to understand:
```
llmchoice = response["text"][:3] if "text" in response else None

if llmchoice in LLM_CHOICE_TO_FLOAT:
	return ScoreResult(score=LLM_CHOICE_TO_FLOAT[llmchoice], category=LLM_CHOICE_TO_CATEGORIES[llmchoice])
else:
	return ScoreResult(score=-1.0, category=LLM_CHOICE_TO_CATEGORIES["default"])
```
Even better if you wanted to merge LLM_CHOICE_TO_FLOAT and LLM_CHOICE_TO_CATEGORIES since they have mostly common keys